### PR TITLE
refactor(perception/occupancy_grid_map_outlier_filter): rework parameters

### PR DESCRIPTION
**Pull request by oguzkaganozt**
## Description

Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by YoshiRi** [comment]
Sorry for super-late reply. :bow:  
Though the changes LGTM within this package, this PR breaks current sample execution due to launch file is not well defined in the user of this outlier filter.
I will convert it to draft for now. Re-open this PR after sample rosbag execution problem is fixed.

**Comment by oguzkaganozt** [comment]
Any updates from rosbag execution problem ? @YoshiRi

**Comment by oguzkaganozt** [comment]
Ready for review @YoshiRi

File: perception/autoware_occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_node.cpp [hunk 1]
```diff
@@ -106,13 +106,13 @@ namespace autoware::occupancy_grid_map_outlier_filter
...
   min_points_and_distance_ratio_ =
-    node.declare_parameter("radius_search_2d_filter.min_points_and_distance_ratio", 400.0f);
-  min_points_ = node.declare_parameter("radius_search_2d_filter.min_points", 4);
-  max_points_ = node.declare_parameter("radius_search_2d_filter.max_points", 70);
+    node.declare_parameter<float>("radius_search_2d_filter.min_points_and_distance_ratio");
+  min_points_ = node.declare_parameter<int>("radius_search_2d_filter.min_points");
+  max_points_ = node.declare_parameter<int>("radius_search_2d_filter.max_points", 70);
   max_filter_points_nb_ =
-    node.declare_parameter("radius_search_2d_filter.max_filter_points_nb", 15000);
+    node.declare_parameter<int>("radius_search_2d_filter.max_filter_points_nb", 15000);
```
Comment (Shin-kyoto) [inline perception/autoware_occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_node.cpp:115]: @oguzkaganozt 
Thank you for your PR.
Why is the default parameter written in both code and config file?

**Review by Shin-kyoto — DISMISSED** [review]

File: perception/autoware_occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_node.cpp (hunk 1 above)
Comment (oguzkaganozt) [reply perception/autoware_occupancy_grid_map_outlier_filter/src/occupancy_grid_map_outlier_filter_node.cpp]: Resolved, please review again

**Comment by YoshiRi** [comment]
@oguzkaganozt cc: @Shin-kyoto @badai-nguyen 
Sorry about my super late reply :bow: 

This package change seems to be okay, but this will cause trouble with `occupancy_grid_based_outlier_filter` node declaration in [`ground_segmentation.launch.py`](https://github.com/autowarefoundation/autoware.universe/blob/b9b0fa9dbb4e141519dc8de13d37827763b481f2/launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py#L330-L346).

I think we need following approach:

- 1. add your config file to `autoware_launch`
- 2. modify `ground_segmentation.launch.py` to look for the config file

I will do 1. first, and want to merge this PR after 2. is solved. ( to prevent autoware crush )

**Comment by YoshiRi** [comment]
@badai-nguyen 
Could you review [this](https://github.com/autowarefoundation/autoware_launch/pull/1137)?

**Comment by oguzkaganozt** [comment]
Hi any updates on this ? @YoshiRi @badai-nguyen

**Comment by oguzkaganozt** [comment]
Hi any updates on this ? @badai-nguyen @YoshiRi

**Comment by YoshiRi** [comment]
I think we need to change related launcher:
https://github.com/YoshiRi/autoware.universe/pull/5/commits/80bfc955a2573e8e74311731162df2c52eaf7d8d

But sorry, I had not tested yet.
If you do not mind, can anyone test with loggin simulator?

**Review by badai-nguyen — CHANGES_REQUESTED** [review]
> This package change seems to be okay, but this will cause trouble with `occupancy_grid_based_outlier_filter` node declaration in [`ground_segmentation.launch.py`](https://github.com/autowarefoundation/autoware.universe/blob/b9b0fa9dbb4e141519dc8de13d37827763b481f2/launch/tier4_perception_launch/launch/obstacle_segmentation/ground_segmentation/ground_segmentation.launch.py#L330-L346).
I think we need following approach:
- 1. add your config file to `autoware_launch`
- 2. modify `ground_segmentation.launch.py` to look for the config file
I will do 1. first, and want to merge this PR after 2. is solved. ( to prevent autoware crush ) 

@oguzkaganozt Thank you for your contribution, 
I think @YoshiRi already left a comment as above, could you confirm and solve 2 as his suggestion?

**Review by YoshiRi — APPROVED** [review]
I fixed launch problem in a6f83ef201c0adada82b7f916499b21a1fd568d3 and
f3d008dd6350379c060124908f9e0a85274773d3.
Tested with my local PC

**Review by badai-nguyen — APPROVED** [review]

**Pull request merged**